### PR TITLE
Placeholder cache

### PR DESCRIPTION
**Pull request by sebastianbenz**
This adds an LRU cache for placeholder calculation. This also introduces the need for a new config level (when the optimizer gets created). We will have to move many of the options to this config level in the future. 

At the moment, it's not possible to completely disable the cache. Not sure if I should add a flag to disable it.

### CODE REVIEW

File: packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js [hunk 1]
```diff
@@ -48,11 +52,28 @@ function escaper(match) {
...
  * * `maxBlurredPlaceholders`: Specifies the max number of blurred images. Defaults to 5.
+ * * `blurredPlaceholdersCacheSize`: Specifies the max number of blurred images to be cached
+ *   to avoid expensive recalculation. Set to 0 if all placeholders should be cached. Defaults
+ *   to 30.
  *
  * Important: blurry image placeholder computation is expensive. Make sure to
  * only use it for static or cached pages.
  */
 class AddBlurryImagePlaceholders {
+  constructor(config) {
```
Comment (andreban) [inline packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: nit: The next two lines could be written as:

```
constructor({blurredPlaceholdersCacheSize = DEFAULT_CACHED_PLACEHOLDERS} = {}) {
```

File: packages/optimizer/lib/transformers/AmpBoilerplateTransformer.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ const V0_CSS_URL = AMP_CACHE_HOST + '/' + V0_CSS;
  * runtime version is specified, v0.css will be inlined.
  */
 class AmpBoilerplateTransformer {
-  constructor(fetch = OneBehindFetch.create()) {
+  constructor(config, fetch = OneBehindFetch.create()) {
```
Comment (andreban) [inline packages/optimizer/lib/transformers/AmpBoilerplateTransformer.js]: nit: I'm wondering if fetch should be a parameter in itself or another option in the config.

Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/AmpBoilerplateTransformer.js]: Yes, that could make sense. We should keep it in mind.

File: packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: that's too much magic for me

**Pull request merged**